src/benchmarklib/benchmark_utils.cpp

```diff
@@ -310,7 +312,8 @@ nlohmann::json EncodingConfig::to_json() const {
 
   nlohmann::json type_mapping{};
   for (const auto& [type, spec] : type_encoding_mapping) {
-    type_mapping[type] = encoding_spec_to_string_map(spec);
+    const auto& type_str = data_type_to_string.left.find(type)->second;
```

Comment:
We know that `type` will probably be in there, but actually, you never know. Use `at()` when you're directly accessing the element of a map, it will throw in out_of_range.